### PR TITLE
Ec slope hint

### LINKED ISSUE #234: EC SLOPE hints
linked to https://github.com/lambdaclass/cairo-vm_in_go/issues/201 issue
once https://github.com/lambdaclass/cairo-vm_in_go/pull/225 has been merged then 

`EC_DOUBLE_SLOPE_V1`
`COMPUTE_SLOPE_V1`

can be implemented

### PR DESCRIPTION
**Pull request by mmsc2**

### CODE REVIEW

File: pkg/hints/ec_hint.go [hunk 1]
```diff
@@ -44,7 +46,7 @@ func BigInt3FromBaseAddr(addr memory.Relocatable, virtual_machine vm.VirtualMach
 	return BigInt3{Limbs: limbs}, nil
 }
 
-func BigInt3FromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
+func EcFromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]:
```suggestion
func EcPointFromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
```

**Review by fmoletta — COMMENTED** [review]
Integration test?

**Comment by mmsc2** [comment]
> Integration test?

I just try but there is still unimplemented hints to do an integration test

File: pkg/hints/ec_hint.go [hunk 2]
```diff
@@ -114,3 +116,98 @@ func ecNegateEmbeddedSecpP(virtual_machine vm.VirtualMachine, exec_scopes types.
...
+%}
+*/
+func computeDoublingSlope(virtual_machine vm.VirtualMachine, exec_scopes types.ExecutionScopes, ids_data hint_utils.IdsManager, point_alias string, secp_p big.Int, alpha big.Int) error {
+	exec_scopes.AssignOrUpdateVariable("SECP_P", secp_p)
+
+	point, err := EcPointFromVarName(point_alias, virtual_machine, ids_data)
+	if err != nil {
+		return err
+	}
+	fmt.Println("after point")
```
Comment (entropidelic) [inline pkg/hints/ec_hint.go]: remove :)

Comment (mmsc2) [reply pkg/hints/ec_hint.go]: done

**Review by jrchatruc — APPROVED** [review]

**Comment by entropidelic** [comment]
A general comment for the whole PR, let's be consistent with the naming conventions, like using camel case and shorter names when applicable (for example, `vm` instead of `virtual_machine` or `virtualMachine`).

**Comment by entropidelic** [comment]
other than the namings, PR looks good 👍

File: pkg/hints/ec_hint.go [hunk 3]
```diff
@@ -44,18 +45,18 @@ func BigInt3FromBaseAddr(addr memory.Relocatable, virtual_machine vm.VirtualMach
 	return BigInt3{Limbs: limbs}, nil
 }
 
-func BigInt3FromVarName(name string, virtual_machine vm.VirtualMachine, ids_data hint_utils.IdsManager) (EcPoint, error) {
-	point_addr, err := ids_data.GetAddr(name, &virtual_machine)
+func EcPointFromVarName(name string, vm VirtualMachine, idsData IdsManager) (EcPoint, error) {
+	point_addr, err := idsData.GetAddr(name, &vm)
```
Comment (entropidelic) [inline pkg/hints/ec_hint.go]: pointAddr

File: pkg/hints/ec_hint.go [hunk 4]
```diff
@@ -66,34 +67,34 @@ func BigInt3FromVarName(name string, virtual_machine vm.VirtualMachine, ids_data
...
 	}
 
-	point_y, err := point.AddInt(3)
+	pointY, err := point.AddInt(3)
 	if err != nil {
 		return err
 	}
 
-	y_bigint3, err := BigInt3FromBaseAddr(point_y, virtual_machine)
+	y_bigint3, err := BigInt3FromBaseAddr(pointY, vm)
```
Comment (entropidelic) [inline pkg/hints/ec_hint.go]: yBigint3

**Review by entropidelic — APPROVED** [review]

**Pull request merged**